Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionSidebar.h

```diff
@@ -0,0 +1,142 @@
+ */
+- (void)webViewWasUpdated;
+
+/*!
+ @abstract Called when the sidebar's title has been updated.
+ @discussion This method is called when the sidebar's title is updated through JavaScript API usage. When this method is called, the browser should query the current sidebar object for the new title in order to display it.
+ */
+- (void)titleWasUpdated;
+
+@end
+
+/*!
+ @abstract This delegate notifies the browser that properties relevant to a currently-closed sidebar have changed.
+ @discussion Implementing this delegate will allow the browser to react to changes to the sidebar object stemming from JavaScript API usage relating to the sidebar's enablement state or icon.
+ */
+WK_CLASS_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA), visionos(WK_XROS_TBA))
+NS_SWIFT_NAME(WKWebExtension.SidebarEnablementDelegate)
+@protocol _WKWebExtensionInactiveSidebarDelegate <NSObject>
+
+/*!
+ @abstract Called when the sidebar has changed from a disabled state to an enabled state.
+ */
+- (void)sidebarWasEnabled;
+
+/*!
+ @abstract Called when the sidebar has changed from an enabled state to a disabled state.
+ */
+- (void)sidebarWasDisabled;
+
+/*!
+ @abstract Called when the sidebar's icon has been updated
+ */
+- (void)iconWasUpdated;
+
+@end
+
+/*!
+ @abstract A `_WKWebExtensionSidebar` object encapsulates the properties for a specific web extension sidebar.
+ @discussion When this property is `nil`, it indicates that the action is the default action and not associated with a specific tab.
+ */
+WK_CLASS_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA), visionos(WK_XROS_TBA))
+NS_SWIFT_NAME(WKWebExtension.Sidebar)
+@interface _WKWebExtensionSidebar : NSObject
+
+- (instancetype)init NS_UNAVAILABLE;
+- (instancetype)new NS_UNAVAILABLE;
+
+/*! @abstract The extension context to which this sidebar is related. */
+@property (nonatomic, readonly, weak) WKWebExtensionContext *webExtensionContext;
+
+/*! @abstract The tab that this sidebar is associated with, or `nil` if it is the default sidebar */
+@property (nonatomic, readonly, nullable, weak) id <WKWebExtensionTab> associatedTab;
+
+/*! @abstract The title of this sidebar. */
+@property (nonatomic, readonly, copy) NSString *title;
+
+/*! @abstract Whether this sidebar is enabled or not. */
+@property (nonatomic, readonly, getter=isEnabled) BOOL enabled;
+
+/*! @abstract The web view which should be displayed when this sidebar is opened. */
+@property (nonatomic, readonly, nullable) WKWebView *webView;
+
+/*! @abstract The delegate for this sidebar which will receive callbacks when the title or web view is updated. */
+@property (nonatomic, nullable, weak) id <_WKWebExtensionActiveSidebarDelegate> activeSidebarDelegate;
+
+/*! @abstract The delegate for this sidebar which will receive callbacks when it is enabled, disabled, or the icon is changed. */
+@property (nonatomic, nullable, weak) id <_WKWebExtensionInactiveSidebarDelegate> inactiveSidebarDelegate;
```

Comment:
A single delegate would be best, if not just use the controller delegate like I mention elsewhere.